src/Orleans.Runtime/Core/GrainRuntime.cs

```diff
@@ -99,11 +101,22 @@ public IStorage<TGrainState> GetStorage<TGrainState>(Grain grain) where TGrainSt
             return new StateStorageBridge<TGrainState>(grainTypeName, grain.GrainReference, grainStorage, this.loggerFactory);
         }
 
-        private static void CheckRuntimeContext()
+        public static void CheckRuntimeContext()
         {
-            if (RuntimeContext.Current == null)
+            var context = RuntimeContext.Current;
+
+            if (context == null)
+            {
+                ThrowMissingContext();
```

Comment:
Minor but I think we should add a comment on why you are declaring a function here

Reply:
It's for the JIT only, to allow inlining. I think once people know this pattern then the comment at every usage is redundant. For those that don't know the pattern, the usage is harmless if a little verbose.

It's mentioned in this post: https://reubenbond.github.io/posts/dotnet-perf-tuning

Reply:
I wonder if some sort of a `ThrowHelper` should be added and extensively used. I also wonder if there's something to grab (otherwise than inspiration) from CoreFx. Just throwing this ideas here.

Reply:
https://github.com/dotnet/corefx/pull/35213 via https://github.com/dotnet/corefx/pull/35213

> Across .NET Core, we’re also paying more attention to assembly size, in particular as it can impact ahead-of-time (AOT) compilation. PRs like dotnet/corefx#35213, which employs “ThrowHelpers” in the heavily-generic LINQ, help to reduce generated code size, which has benefits in and of itself but can also help with other areas of performance.